pkg/operator/validations.go

```diff
@@ -34,3 +35,19 @@ func ValidateDurationField(durationField string) error {
 	}
 	return nil
 }
+
+// ValidateScrapeTimeout validates value of scrapeTimeout based on scrapeInterval
+func ValidateScrapeTimeout(scrapeTimeout, scrapeInterval string) error {
+	// since default scrapeInterval set by operator is 1m
+	if scrapeInterval == "" {
+		scrapeInterval = "60s"
```

Comment:
We could set this via OpenAPI in https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/types.go#L174 via an annotation:

`// +kubebuilder:default:=60s`

What do you think?

Reply:
The purpose of checking if scrapeTimeout > 60s was incase if scrapeInterval is not specified operator sets to `60s` per https://github.com/prometheus-operator/prometheus-operator/blob/9c0db5656f04e005de6a0413fd8eb8f11ec99757/pkg/apis/monitoring/v1/types.go#L173

I am checking where it sets this value by default

Reply:
It actually sets `30s` by default, I could confirm that in minikube cluster as well. 

https://github.com/prometheus-operator/prometheus-operator/blob/5f72fbc655b416b9b50ede04ed2c0a64cda22933/pkg/prometheus/promcfg.go#L339

@simonpasquier We need to correct this in `types.go` comments since it  shows in `kubectl explain` ? or Am I missing something?

Reply:
Like @PhilipGough  suggested we can set this via OpenAPI instead of setting here

Reply:
I personally think we should make the OpenAPI spec the source of truth to avoid situations where we have drift between the docs and the code like you have found here.

If @simonpasquier agrees with that sentiment we can just do it for `scrapeInterval ` as part of this PR and create an issue to investigate and tackle other scenarios.

Reply:
I agree, lets define the default value of scrapeInterval in the OpenAPI spec. We can also add a default value for scrapeTimeout.

Reply:
@simonpasquier @PhilipGough  I tested with `//+kubebuilder:default="30s"` for interval and `//+kubebuilder:default="`10s"` for timeout. But it requires updating most of the unit tests since this default value will be set runtime as I understand


https://github.com/prometheus-operator/prometheus-operator/blob/5f72fbc655b416b9b50ede04ed2c0a64cda22933/pkg/prometheus/promcfg_test.go#L97-L103

Can we do that in separate PR by handling all the default values in openapi spec as possible?